xbmc/cores/VideoPlayer/VideoRenderers/HwDecRender/VideoLayerBridgeDRMPRIME.cpp

```diff
@@ -140,6 +166,14 @@ void CVideoLayerBridgeDRMPRIME::Configure(CVideoBufferDRMPRIME* buffer)
     m_DRM->AddProperty(connector, "content type", m_DRM->GetContentType(true));
     m_DRM->SetActive(true);
   }
+
+  struct plane* plane = m_DRM->GetPrimaryPlane();
+  if (m_DRM->HasProperty(plane, "COLOR_ENCODING") &&
+      m_DRM->HasProperty(plane, "COLOR_RANGE"))
+  {
+    m_DRM->AddProperty(plane, "COLOR_ENCODING", GetColorEncoding(buffer));
+    m_DRM->AddProperty(plane, "COLOR_RANGE", GetColorRange(buffer));
+  }
```

Comment:
Not sure if I understand this code?
You check if the property is there and if it is you add another one?

Reply:
@a1rwulf we need to check if the drm driver has the property available. If it is available then we can set the property. Maybe the wording is just odd.

Reply:
yeah maybe I'm confused because we have AddProperty and SetProperty in our public API in DRMUtils.

Reply:
Maybe we can make DRMLegacy override AddProperty instead of SetProperty and just remove the SetProperty method?

Reply:
Ah, it looks like it's just wording between legacy (`drmModeObjectSetProperty`) and atomic (`drmModeAtomicAddProperty`) operations. Maybe we can clean that up at some point.

Reply:
`HasProperty` checks if the plane supports the property or not and the `AddProperty` adds a new property value to the next commit.
Should I rename `HasProperty` to `SupportsProperty` to make it less confusing?

The reason for the `HasProperty` check is to not trigger warnings when the plane don't support the property.

Reply:
That would be nice.
Still we could also cleanup SetProperty vs. AddProperty as well.